merino/middleware/featureflags.py

```diff
@@ -0,0 +1,16 @@
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.requests import Request
+
+from merino.featureflags import session_id_context
+
+
+class FeatureFlagsMiddleware(BaseHTTPMiddleware):
+    """
+    Sets the a ContextVar for session_id so that it can be used
+    to consistently bucket flags within a search session
+    """
+
+    async def dispatch(self, request: Request, call_next):
+        session_id = request.query_params.get("sid")
+        session_id_context.set(str(session_id))
```

Comment:
```suggestion
        session_id_context.set(session_id)
```

`request_params.get("sid")` is already a str if present.  If `sid` is not set in `request.query_params`, `str(session_id)` will produce a string `"None"` which is probably not what we want to set for `session_id_context`.